packages/atomic/src/components/atomic-result-list/atomic-result-list.tsx

```diff
@@ -92,9 +92,10 @@ export class AtomicResultList implements InitializableComponent {
   }
 
   private get results() {
+    const timestamp = Date.now();
     return this.resultListState.results.map((result) => (
       <atomic-result
-        key={result.raw.permanentid}
+        key={`${result.raw.permanentid}${timestamp}`}
```

Comment:
Could we use `searchUID` from `engine.state.search.response.searchUid` instead ?
This should change on any new actual new search performed.

I have a gut feeling that this timestamp would cause problem in a server side rendered scenario 🤔 

Should there be an accompanying end to end test for this ?

Reply:
We could but it would require adding this piece of state to the result list controller

Reply:
Could we remove the `key` attribute?

The `key` is an optimization to allow Stencil/React to avoid rerendering the same DOM element. However, here, we always want to rerender the result since the same result appears differently depending on the query.

Adding a `timestamp`, `searchUid`, or any changing value, is by-passing the optimization. We may as well remove the `key`.

If Stencil shows a warning, or this creates a performance hit (e.g. with 50 or 100 results per page), then the next best would be add the `query` to the key. The `key` should contain the dependencies that affect the rendered output.

Reply:
Not putting the key also doesn't re-render, stencil works a bit differently than react in this manner it seems.
> the next best would be add the query to the key
Not only the query affects the render content, what if a field(facet) is selected and a condition inside the result or a value is affected?

Best is the searchUid which is missing from the controller, next best is something like a Date.now/Math.random

Reply:
Ok, since we cannot deactivate the optimization, and there's more than just the query affecting the highlights, then let's add the `searchUid` in there.